.github/scripts/generate_ascii_report.py

```
@@ -0,0 +1,83 @@
+    bars = '▁▂▃▄▅▆▇█'
+    max_val = max(data) if data else 1
+    min_val = min(data) if data else 0
+    delta = max_val - min_val if max_val > min_val else 1
+    
+    sparkline = ''
+    for val in data:
+        index = int((val - min_val) / delta * (len(bars) - 1))
+        sparkline += bars[min(index, len(bars) - 1)]
+    return sparkline
+
+def main():
```

Flake8 (7.2.0)

[error] expected 2 blank lines, found 1

(E302)